src/java.base/share/classes/java/util/zip/DeflaterInputStream.java

```diff
@@ -34,6 +34,19 @@
  * Implements an input stream filter for compressing data in the "deflate"
  * compression format.
  *
+ * <h2 id="deflater-usage">Deflater Usage</h2>
+ * <p>This class uses a {@link Deflater} for compressing the data. Two forms of constructors
+ * are available for constructing a {@code DeflaterInputStream} - one which accepts
+ * a {@code Deflater} and one which doesn't. The constructors that don't accept a {@code Deflater}
```

Comment:
> `The constructors that don't accept a {@code Deflater}`

Regarding the internal `Deflater` that will be created and its life cycle will be managed by the `DIS`... This sounds like an _internal implementation detail_ and `DIS` takes care of this internal resource nicely. IMHO, providing this information brings no benefit to users and sort of hand-cuffs ourselves to this behavior. Shall we remove that part from the docs? (Note that this remark applies to all 4 classes whose JavaDoc is updated.)